processor/src/main/java/org/mapstruct/ap/internal/model/source/TargetReference.java

```diff
@@ -142,25 +148,26 @@ public TargetReference build() {
             // 3. @MappingTarget, with
             // 4. or without parameter name.
             String[] targetPropertyNames = segments;
-            List<PropertyEntry> entries = getTargetEntries( resultType, targetPropertyNames, false );
+            List<PropertyEntry> entries = getTargetEntries( resultType, targetPropertyNames );
             foundEntryMatch = (entries.size() == targetPropertyNames.length);
             if ( !foundEntryMatch && segments.length > 1
                 && matchesSourceOrTargetParameter( segments[0], parameter, reverseSourceParameter, isReverse ) ) {
                 targetPropertyNames = Arrays.copyOfRange( segments, 1, segments.length );
-                entries = getTargetEntries( resultType, targetPropertyNames, false );
+                entries = getTargetEntries( resultType, targetPropertyNames );
                 foundEntryMatch = (entries.size() == targetPropertyNames.length);
             }
 
-            if ( !foundEntryMatch ) {
+            if ( !foundEntryMatch && errorMessage != null) {
```

Comment:
what message will be printend if `foundEntryMatch==false` and `errorMessage==null`?